index.js

```diff
@@ -345,54 +346,61 @@ Client.prototype._resolveToken = function (cb) {
 
     done()
   })
+  .catch(err => {return done(err)})
```

Comment:
What errors is this supposed to be catching?

Reply:
3 types of errors are caught:
1. The request was made and the server responded with a status code that falls out of the range of 2xx (error.response holds information)
2. The request was made but no response was received (error.request holds the information)
3. Something happened in setting up the request that triggered an Error (error.message could give more information)

Reply:
All of those are something axios would handle internally though right? Unless I am mistaken the abort controller is intended to send an abort signal into axios when things outside of the request itself go wrong and we need to abort the request.

Reply:
I think, abort is slightly different solution where user chooses to cancel the request.
In this case, we get a handle for cancelling the in-progress request. If we cancel the request, say for a timeout scenario, some debug information is not obtained as we get in the catch block.